pkg/ebpf/tracee.go

```diff
@@ -59,6 +60,45 @@ type fileExecInfo struct {
 	Hash      string
 }
 
+// fileUniqueId is an unique identifier for a file, based on its device and inode.
+type fileUniqueId struct {
```

Comment:
I guess you confirmed that this keying works across containers? So two busybox containers will share inode+device for its `/bin/busybox` files?

Reply:
Good point. I finished it late at night so I didn't check it until now - I'm still sleepy 😴.

Unfortunately the key dev/inode won't work, since each new container has its own mountns (device).

But the good news is that the inode is the same. So we could key the hash with `origin/inode`, origin being `host` or `ImageDigest`

- `"host,inode"`
- `"alpine@sha256:34871e7290500828b39e22294660bee86d966bc0017544e848dd9a255cdf59e0,inode"`

The unique **minor** problem is that `ImageDigest` is only available on the second event arrived from that container, so the hash shouldn't be computed for that case.

WDYT?

```
# alpine 1

FROM Args: dev: 88, ino: 9580902
ContainerID: fd19818e0560b652e09ab923dc26cc48d91947e1cc6c961bf2939a26033a9a13
ImageDigest: alpine@sha256:34871e7290500828b39e22294660bee86d966bc0017544e848dd9a255cdf59e0
fileId: &{dev:88 ino:9580902}
08:30:54:354089  fd19818e0560      fd19818e0560  4026533824   4026533828   0      uname            15      15      1       0                sched_process_exec        cmdpath: /bin/uname, pathname: /bin/busybox, dev: 88, inode: 9580902, ctime: 1701897542997954943, inode_mode: 33261, interpreter_pathname: <nil>, interpreter_dev: <nil>, interpreter_inode: <nil>, interpreter_ctime: <nil>, argv: [uname], interp: /bin/uname, stdin_type: S_IFCHR, stdin_path: /dev/pts/0, invoked_from_kernel: 0, env: <nil>, sha256: 8c6864b97c2b9ff892c859c59363482cb19a546a33a204b83a24f29048b2118f

#alpine 2

FROM Args: dev: 48, ino: 9580902
ContainerID: 1e804fb5bc6bbe0704149564dfb24b372068b4e96cd2f04de37df0ebd7e2fcbc
ImageDigest: alpine@sha256:34871e7290500828b39e22294660bee86d966bc0017544e848dd9a255cdf59e0
fileId: &{dev:48 ino:9580902}
08:31:07:491494  1e804fb5bc6b      1e804fb5bc6b  4026533219   4026533223   0      uname            14      14      1       0                sched_process_exec        cmdpath: /bin/uname, pathname: /bin/busybox, dev: 48, inode: 9580902, ctime: 1701897542997954943, inode_mode: 33261, interpreter_pathname: <nil>, interpreter_dev: <nil>, interpreter_inode: <nil>, interpreter_ctime: <nil>, argv: [uname], interp: /bin/uname, stdin_type: S_IFCHR, stdin_path: /dev/pts/0, invoked_from_kernel: 0, env: <nil>, sha256: 8c6864b97c2b9ff892c859c59363482cb19a546a33a204b83a24f29048b2118f
```

Reply:
By the way, I am stating for the device/inode while should be just getting those from `event.Args`. Gonna change that.

Reply:
And another question - from what I know device ID is not related to mount namespaces.
From what I know, it just changes the mounting points, but devices are kernel structures.
I am not 100% sure of myself though, I just curios to know from where have you brought this information.

Reply:
@AlonZivony take a look at this new approach which let the choice for the user: f24813f12